File: src/core/interfaces/types/loot.ts
/**
 * Base item types and interfaces for the loot system
 */

export enum ItemType {
  WEAPON = 'WEAPON',
  ARMOR = 'ARMOR',
  POTION = 'POTION',
  SCROLL = 'SCROLL',
  MATERIAL = 'MATERIAL',
  TREASURE = 'TREASURE',
  KEY = 'KEY',
  QUEST = 'QUEST',
  MISC = 'MISC'
}

export enum ItemRarity {
  COMMON = 'COMMON',
  UNCOMMON = 'UNCOMMON',
  RARE = 'RARE',
  EPIC = 'EPIC',
  LEGENDARY = 'LEGENDARY'
}

export interface BaseStats {
  damage?: number;
  armor?: number;
  healing?: number;
  duration?: number;
  charges?: number;
  [key: string]: number | undefined;
}

export interface BaseItem {
  id: string;
  name: string;
  description: string;
  type: ItemType;
  weight: number;
  value: number;
  baseStats: BaseStats;
  createdAt: Date;
  updatedAt: Date;
}

export interface RarityTier {
  id: number;
  name: ItemRarity;
  probability: number;
  valueMultiplier: number;
  colorHex: string;
}

export interface ItemWithRarity extends BaseItem {
  rarity: RarityTier;
} 

File: backend/app/src/core/interfaces/types/loot.py
from typing import Any, Union
from enum import Enum


/**
 * Base item types and interfaces for the loot system
 */
class ItemType(Enum):
    WEAPON = 'WEAPON'
    ARMOR = 'ARMOR'
    POTION = 'POTION'
    SCROLL = 'SCROLL'
    MATERIAL = 'MATERIAL'
    TREASURE = 'TREASURE'
    KEY = 'KEY'
    QUEST = 'QUEST'
    MISC = 'MISC'
class ItemRarity(Enum):
    COMMON = 'COMMON'
    UNCOMMON = 'UNCOMMON'
    RARE = 'RARE'
    EPIC = 'EPIC'
    LEGENDARY = 'LEGENDARY'
class BaseStats:
    damage?: float
    armor?: float
    healing?: float
    duration?: float
    charges?: float
    [key: Union[str]: float, None]
class BaseItem:
    id: str
    name: str
    description: str
    type: \'ItemType\'
    weight: float
    value: float
    baseStats: \'BaseStats\'
    createdAt: Date
    updatedAt: Date
class RarityTier:
    id: float
    name: \'ItemRarity\'
    probability: float
    valueMultiplier: float
    colorHex: str
class ItemWithRarity:
    rarity: \'RarityTier\' 